roles/validate/files/rules/required_rules/502_policy_vrf_lite_cross_reference.py

```diff
@@ -0,0 +1,313 @@
+2.  BGP and OSPF should be defined in two different policy.
+    vxlan.overlay_extensions.vrf_lites.ospf
+    vxlan.overlay_extensions.vrf_lites.bgp
+    vxlan.overlay_extensions.vrf_lites.switches.bgp
+    vxlan.overlay_extensions.vrf_lites.switches.bgp_peers
+    vxlan.overlay_extensions.vrf_lites.switches.interfaces.ospf
+3.  route_reflector_client for AF IPv4 and IPv6 config should not be allowed.
+    vxlan.overlay_extensions.vrf_lites.switches.bgp_peers.address_family_ipv4_unicast.route_reflector_client
+    vxlan.overlay_extensions.vrf_lites.switches.bgp_peers.address_family_ipv6_unicast.route_reflector_client
+4.  Check if OSPF Authentication is enabled, key must be provided
+    vxlan.overlay_extensions.vrf_lites.switches.interfaces.ospf.auth_key
+5.  Check AREA 0 is standard
+    vxlan.overlay_extensions.vrf_lites.ospf.areas.area_type
+6.  Check static routes across Policies
+    vxlan.overlay_extensions.vrf_lites.switches.static_routes
+7.  Check ospf network type for Loopback
+    vxlan.overlay_extensions.vrf_lites.switches.interfaces.ospf.network_type
+'''
+
+
+class Rule:
+    id = "502"
+    description = "Verify VRF-Lites Cross Reference Between Policies, Groups, and Switches"
+    severity = "HIGH"
+    results = []
+
+    @classmethod
+    def match(cls, data):
+        '''
+        function used by iac-validate
+        '''
+        vrf_lites = []
+        topology_switches = []
+        policies = []
+        switch_policy = []
+        static_routes_compliance = []
+
+        # Get fabric switches
+        if data.get("vxlan"):
+            if data["vxlan"].get("topology"):
+                if data.get("vxlan").get("topology").get("switches"):
+                    topology_switches = data.get("vxlan").get("topology").get("switches")
+
+        # Get vrf-lites policies
+        if data.get("vxlan", None):
+            if data["vxlan"].get("overlay_extensions", None):
+                if data["vxlan"].get("overlay_extensions").get("vrf_lites", None):
+                    vrf_lites = data["vxlan"]["overlay_extensions"]["vrf_lites"]
+
+        for policy in vrf_lites:   # Check Global Level
+            if policy.get('name', None):
+                policies.append(policy['name'])
+                if policy.get('ospf'):  # Check OSPF parameters
+                    if policy['ospf'].get('process'):
+                        pass
+                    else:
+                        cls.results.append(
+                            f"vxlan.overlay_extensions.vrf_lites.{policy['name']}.ospf.process. "
+                            f"OSPF process is not defined."
+                )
+                cls.check_global_ospf_and_bgp(policy)
+                cls.check_global_ospf_area(policy)
+
+                if policy.get('switches'):  # Check switch Level
+                    for switch_policy in policy['switches']:
+                        cls.check_switch_level(switch_policy, policy,
```

Comment:
Lets update the comment to check switch level and update static_routes_compliance if static_routes exist in switch policy for use in later compliance checks.

Reply:
done